src/proxy/sessions/session.rs

```diff
@@ -169,20 +163,20 @@ impl Session {
                                 tracing::error!(%error, %source, dest = ?endpoint, "Error receiving packet");
                             },
                             Ok((size, recv_addr)) => {
-                                metrics.rx_bytes_total.inc_by(size as u64);
-                                metrics.rx_packets_total.inc();
+                                crate::metrics::PACKETS_SIZE.with_label_values(&[crate::metrics::WRITE_DIRECTION_LABEL]).set(size as f64);
```

Comment:
Feel free to correct me if I'm missing something, but a gauge seems more useful for packet size because you can still get total number of bytes by calling `sum` in something like Grafana, while also being able to get other data, for example; getting the average/median packet size.

Reply:
Right, but the value of that guage is only scraped by prometheus at a given interval. So if you want to `sum()`, all you are going to get is the sum of the current byte size every, say 30s, which is not an accurate sum of all packets that have gone through the system.

Guages are useful when you have a value that can arbitrarily go up and down. A counter should be used for a value that is ever increasing.

One can graph at the rate of change over time with the total amount to see spikes in byte traffic:
> rate(v range-vector) calculates the per-second average rate of increase of the time series in the range vector.
https://prometheus.io/docs/prometheus/latest/querying/functions/#rate

I haven't played with them, but there are also functions such as:
> avg_over_time(range-vector): the average value of all points in the specified interval.
https://prometheus.io/docs/prometheus/latest/querying/functions/#aggregation_over_time

That can also give you queries over the entire data set.

Reply:
Ah, that makes sense, changed it back to a counter.